### PR TITLE
Compact Tally Reports

### LINKED ISSUE #359: Tabulation reports should be more compact
lots of wasted space on full sheet of paper right now.

Maybe 2 or 3 column design? Header that shows total number of ballots cast across whole width and then 2 or 3 columns of per-contest info?

### PR DESCRIPTION
**Pull request by beausmith**
Resolves #359 

image

### CODE REVIEW

File: apps/election-manager/src/App.test.tsx [hunk 1]
```diff
@@ -279,7 +278,8 @@ it('tabulating CVRs', async () => {
   // When there are no CVRs imported the full tally report is labeled as the zero report
   fireEvent.click(getByText('View Unofficial Full Election Tally Report'))
   // Verify the zero report generates properly
-  expect(getByTestId('tally-report-contents')).toMatchSnapshot()
+  // TODO: Do this without a snapshot.
+  // expect(getByTestId('tally-report-contents')).toMatchSnapshot()
```
Comment (benadida) [inline apps/election-manager/src/App.test.tsx:282]: maybe we should keep this test until we find an alternative?

File: apps/election-manager/src/App.test.tsx [hunk 2]
```diff
@@ -335,7 +335,6 @@ it('tabulating CVRs with SEMS file', async () => {
   const totalRow = domGetByTestId(ballotsByVotingMethod[0], 'total')
   domGetByText(totalRow, 'Total Ballots Cast')
   domGetByText(totalRow, '200')
-  expect(getByTestId('tally-report-contents')).toMatchSnapshot()
```
Comment (benadida) [inline apps/election-manager/src/App.test.tsx:338]: same comment --> keep until we have alternative? I don't think the snapshot on the report is all that bad, I don't think it will change willy-nilly with unrelated changes.

**Review by benadida — APPROVED** [review]

File: apps/election-manager/src/App.test.tsx (hunk 1 above)
Comment (beausmith) [reply apps/election-manager/src/App.test.tsx:282]: Alternative found!
```
  // Verify the zero report generates properly
  expect(getAllByText('0').length).toBe(40)
```

File: apps/election-manager/src/App.test.tsx (hunk 2 above)
Comment (beausmith) [reply apps/election-manager/src/App.test.tsx:338]: What is this line testing for? How does the dev know if what changed was supposed to change?

Without a clear assertion the value of this test is not discernible.

File: apps/election-manager/src/components/TallyReportSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  ballotCountsByVotingMethod,
+}) => {
+  const totalBallotCount = internalBallotCount + (externalBallotCount ?? 0)
+
+  return (
+    <BallotSummary>
+      <h3>Ballots by Voting Method</h3>
+      <Table data-testid="voting-method-table">
+        <tbody>
+          {Object.keys(ballotCountsByVotingMethod).map((votingMethod) => {
```
Comment (eventualbuddha) [inline apps/election-manager/src/components/TallyReportSummary.tsx:42]: This feels like it's begging to be a `Map` or an array of objects.

**Review by eventualbuddha — APPROVED** [review]

File: apps/election-manager/src/App.test.tsx (hunk 2 above)
Comment (beausmith) [reply apps/election-manager/src/App.test.tsx:338]: Overall this repo does not yet have 100% test coverage and when we do, we should not rely on snapshots to blanket cover code which we should be looking for specific values or which have comments defining what the snapshot is checking for such that future devs don't have to figure out what the test is for each time they run in to it. With large snapshots, I fear that a dev will simply update the snapshot and move on (I have! 😛). 

I will add these testing lines back, because there is also no harm in doing so… but with comments noting that large snapshots without clear understanding of what is being checked for are not useful.

File: apps/election-manager/src/components/TallyReportSummary.tsx (hunk 3 above)
Comment (beausmith) [reply apps/election-manager/src/components/TallyReportSummary.tsx:42]: Would like to refactor but it's outside the scope of this PR. 🙃

**Review by carolinemodic — APPROVED** [review]
excited for this!
